### PR TITLE
[Drop Pin] Render pin using default renderer instead of using imageview

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards #1352 

 * Fixes FeatureClusterManager when removing unused geometries
 * Render pin generically instead of creating bitmap directly
 * Cleanup layout

@scolsen  PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/map/gms/FeatureClusterManager.kt [hunk 1]
```diff
@@ -53,13 +53,11 @@ class FeatureClusterManager(context: Context?, map: GoogleMap) :
 
   /** Remove a set of features from this manager's clusters. */
   fun removeLocationOfInterestFeatures(features: Set<Feature>) {
-    val existingIds = algorithm.items.map { it.feature.id }.toSet()
-    val deletedIds = existingIds.intersect(features.map { it.id }.toSet())
-    val deletedPoints: Set<FeatureClusterItem> =
-      algorithm.items.filter { deletedIds.contains(it.feature.id) }.toSet()
+    val deletedIds = algorithm.items.map { it.feature.id } - features.map { it.id }.toSet()
```
[the remark this answers is not in the document]
Comment (scolsen) [reply ground/src/main/java/com/google/android/ground/ui/map/gms/FeatureClusterManager.kt:56]: e.g. given `algo.items = {A,B,C}` and `features={A,B,D}` deleted items == `{C}` and calling this method will remove C from the items but not A or B.

[the remark this answers is not in the document]
Comment (scolsen) [reply ground/src/main/java/com/google/android/ground/ui/map/gms/FeatureClusterManager.kt:56]: which, this implementation probably makes more sense given how the DB works -- I would just rename this to `removeStale` or something similar as you did for the test to make it clearer what's going on.

File: ground/src/main/java/com/google/android/ground/ui/map/gms/FeatureClusterManager.kt [hunk 2]
```diff
@@ -53,13 +53,11 @@ class FeatureClusterManager(context: Context?, map: GoogleMap) :
 
   /** Remove a set of features from this manager's clusters. */
   fun removeLocationOfInterestFeatures(features: Set<Feature>) {
```
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/map/gms/FeatureClusterManager.kt]: Done

File: ground/src/main/java/com/google/android/ground/ui/map/gms/FeatureClusterManager.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/map/gms/FeatureClusterManager.kt:56]: Done

**Review by gino-m — APPROVED** [review]

**Pull request merged**